### PR TITLE
feat: add rule for branch name

### PR DESCRIPTION
**Pull request by nejdetkadir**
# Checklist

- [x] I've reviewed my code. (\*)
- [ ] Does this task have an ID?
- [x] Should this task be tested by testers? (\*)
- [ ] Is there a need to add new translations
  - [ ] I've placed all translations in the appropriate JSON files. (\*)
  - [ ] I've verified that the translations work as expected in different languages. (\*)
- [ ] Are there any UI changes?
  - [ ] Screenshots or screen recordings of the changes have been attached to the PR. (\*)
- [x] No unnecessary console logs or commented out code remains. (\*)
- [x] I didn't hard-code any sensitive information.(\*)
- [x] I've tested browser compatibility. (e.g. Chrome, Safari, Mozilla, Opera) (\*)
- [x] All new and existing tests have passed. (\*)
- [x] I've checked for potential performance issues. (e.g. slow component mounting, too many HTTP requests, unnecessary re-renders) (\*)
- [x] I've addressed accessibility concerns. (\*)
- [ ] I've added a new dependency.
  - [ ] I've checked that all newly added dependencies are secure and up-to-date. (\*)
- [x] The PR contains only changes related to the task at hand. If there are any changes outside of the task, I've mentioned them in the description. (\*)

---

**Related Issue Link(s):** ...

**Description:** ...

### CODE REVIEW

**Comment by nejdetkadir** [comment]
BOT MESSAGE :robot:


All good for checklist :green_circle:

File: src/constants/inputKeys.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ const INPUT_KEYS = Object.freeze({
   ASSIGNEE_REQUIRED: 'assigneeRequired',
   CHECKLIST_REQUIRED: 'checklistRequired',
   SEMANTIC_TITLE_REQUIRED: 'semanticTitleRequired',
+  SEMANTIC_PR_TITLE_REQUIRED: 'semanticPRTitleRequired',
```
Comment (sonergonencler) [inline src/constants/inputKeys.ts:6]: bunu branch name kontrolü için mi koyduk, pr title'ı için mi koyduk

SEMANTIC_BRANCH_NAME_REQUIRED
SEMANTIC_PR_TITLE_REQUIRED
  
 böyle ilerletsek daha iyi değil mi

File: src/hooks/useInputs.ts [hunk 2]
```diff
@@ -27,13 +28,18 @@ export default function useInputs(): UseInputsReturnTypes {
     core.getInput(INPUT_KEYS.SEMANTIC_TITLE_REQUIRED, { required: true })
   );
 
+  const isSemanticPRTitleRequired = Boolean(
```
Comment (sonergonencler) [inline src/hooks/useInputs.ts:31]:
```suggestion
  const isSemanticBranchNameRequired = Boolean(
```

burasını ve kodun devamını branch name kontrolü diye devam ettirsek daha iyi değil mi

**Comment by shft-tech** [comment]
BOT MESSAGE :robot:


All good for checklist :green_circle:

**Review by sonergonencler — APPROVED** [review]

**Pull request merged**